### ISSUE TITLE
Upgrade to Spring Boot 3 RC1

### ISSUE THREAD

**Issue by tstephen** [odrotbohm/spring-restbucks#115]
I made a fork here:https://github.com/tstephen/spring-restbucks/tree/hacking/boot3

The changes were generally very small but I've hit a wall now:
1. If I add back the spring-native 0.12.1 dependency (there does not appear to be an updated one to work with Boot 3) then fails with:
    ```
    [1/7] Initializing...
    (0.0s @ 0.37GB) 
    Error: Could not find target method: private org.springframework.boot.Banner org.springframework.nativex.substitutions.boot.Target_SpringApplicationBannerPrinter.getImageBanner(org.springframework.core.env.Environment)
    Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
     ```
     which seems a very strange thing to fail on.
3. If I remove the hints and make the `DrinksOptions.BY_NAME` a local var rather than a static field I can delete the `NativeConfiguration` class and progress further. In this case `mvn clean -Pnative native:compile` fails with this error.
    ```
    [2/7] Performing analysis...  [**]                                                                      (61.5s @ 2.20GB)
      29,299 (95.33%) of 30,733 classes reachable
      45,506 (71.31%) of 63,818 fields reachable
     130,967 (72.31%) of 181,123 methods reachable
       1,572 classes,   552 fields, and 6,453 methods registered for reflection
    
    Fatal error: java.lang.TypeNotPresentException: Type kotlinx.serialization.BinaryFormat not present
    ```

Any thoughts on what to try next?

**Comment by odrotbohm** [comment]
I'm sorry you ran into this. The native setup has been in flux for quite a while. Would you mind retrying on the latest `main` and simply run `mvn -Pnative`. That works fine for me.

**Comment by tstephen** [comment]
Yes, working on `main`,  `mvn  -Pnative` and `mvn  -Pnative spring-boot:build-image` both now work fine.

But only the first time. Second build fails (fast) with: 
```
[ERROR] Failed to execute goal net.bytebuddy:byte-buddy-maven-plugin:1.12.18:transform (default) on project restbucks: Failed to transform class files in /mnt/data/git/spring-restbucks/server/target/classes: Cannot resolve type description for org.springframework.data.rest.webmvc.halexplorer.HalExplorer -> [Help 1]
```

**Comment by tstephen** [comment]
Testing against commit `a16bcabdf93b71ba3c94123aac0685d502f5dcca`

**Comment by odrotbohm** [comment]
That's an issue in the jMolecules ByteBuddy plugin. Building with `clean` should do the trick. I'll investigate further.

**Comment by odrotbohm** [comment]
The upstream bug is [here](https://github.com/xmolecules/jmolecules-integrations/issues/190). Fixed in jMolecules ByteBuddy 0.17, i.e. with be578349b3344e08dd7da87fa15de742b7360649.
